### PR TITLE
Helm chart for the MatterMost plugin

### LINKED ISSUE #498: Implement Helm chart for `teleport-plugin-mattermost`
- [x] Write tests (using `helm-unittest`)
- [ ] Add it to the Drone build (build + upload to the staging and the production buckets)
- [ ] Test the Helm chart by installing it and connecting it somewhere (eg. `access-dev.teleportinfra.dev`)
- [ ] Document settings
- [ ] Document installation

### PR DESCRIPTION
**Pull request by kbence**
This change adds a basic Helm chart and documentation for the MatterMost plugin

Closes #498

### CODE REVIEW

File: charts/access/mattermost/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 1.0.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "9.0.0"
```
Comment (wadells) [inline charts/access/mattermost/Chart.yaml:24]: 9.0.4

File: charts/access/mattermost/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    <td>string</td>
+    <td><code>""</code></td>
+    <td>yes</td>
+  </tr>
+  <tr>
+    <td><code>teleport.identitySecretName</code></td>
+    <td>Name of the Kubernetes secret that contains the credentials for the connection</td>
+    <td>string</td>
+    <td><code>""</code></td>
+  </tr>
```
Comment (wadells) [inline charts/access/mattermost/README.md:30]: This row and all below are missing a <td>for the final "Required" column.

File: charts/access/mattermost/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  <tr>
+    <td><code>mattermost.token</code></td>
+    <td>Token to be used to authenticate with Mattermost</td>
+    <td>string</td>
+    <td><code>""</code></td>
+  </tr>
+  <tr>
+    <td><code>mattermost.recipients</code></td>
+    <td>Array of the recipients the plugin should send emails.</td>
```
Comment (wadells) [inline charts/access/mattermost/README.md:52]:
```suggestion
    <td>Array of the recipients the plugin should send access requests to.</td>
```

File: charts/access/mattermost/tests/configmap_test.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+tests:
+  - it: should match the snapshot
+    set:
+      teleport:
+        address: teleport.example.com:1234
+      mattermost:
+        url: https://my.mattermost.com
+        token: test-mattermost-token
+        recipients:
+          - security@mycompany.com
```
Comment (wadells) [inline charts/access/mattermost/tests/configmap_test.yaml:13]: Best to stick to example.com from https://tools.ietf.org/search/rfc2606#section-3.  mycompany.com is a real domain who probably doesn't want this mail.

File: charts/access/mattermost/values.schema.json [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+        "log"
+    ],
+    "properties": {
+        "image": {
+            "$id": "#/properties/image",
+            "type": "object",
+            "default": {},
+            "examples": [
+                {
+                    "repository": "146628656107.dkr.ecr.us-west-2.amazonaws.com/gravitational/teleport-plugin-mattermost",
```
Comment (wadells) [inline charts/access/mattermost/values.schema.json:27]: Use the new quay.io repo.

File: charts/access/mattermost/values.schema.json [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+            "required": [
+                "repository",
+                "pullPolicy",
+                "tag"
+            ],
+            "properties": {
+                "repository": {
+                    "$id": "#/properties/image/properties/repository",
+                    "type": "string",
+                    "default": "146628656107.dkr.ecr.us-west-2.amazonaws.com/gravitational/teleport-plugin-mattermost",
```
Comment (wadells) [inline charts/access/mattermost/values.schema.json:41]: Use the new quay.io repo.

File: charts/access/mattermost/values.schema.json [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+                    "examples": [
+                        "146628656107.dkr.ecr.us-west-2.amazonaws.com/gravitational/teleport-plugin-mattermost"
```
Comment (wadells) [inline charts/access/mattermost/values.schema.json:43]: Use the new quay.io repo.

File: charts/access/mattermost/values.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
+# Default values for mattermost.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# replicaCount: 1
+
+image:
+  repository: 146628656107.dkr.ecr.us-west-2.amazonaws.com/gravitational/teleport-plugin-mattermost
```
Comment (wadells) [inline charts/access/mattermost/values.yaml:8]: Use the new quay.io repo.

**Review by wadells — COMMENTED** [review]
Same comment as #494: This needs to be tied into the general Makefile as well as the CI you set up in https://github.com/gravitational/teleport-plugins/pull/459.

File: charts/access/mattermost/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
+name: teleport-plugin-mattermost
+description: A Helm chart for Kubernetes
```
Comment (logand22) [inline charts/access/mattermost/Chart.yaml:3]: Please update the description.

**Review by wadells — APPROVED** [review]
:ship:

**Review by Joerger — APPROVED** [review]

**Pull request merged**